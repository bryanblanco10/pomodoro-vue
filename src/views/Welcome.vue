<template>
  <section class="welcome">
    <div class="aplication__content welcome__content">
      <h1 class="welcome__title">Hola, ¡bienvenido a Pomodoro!</h1>
      <h2 class="welcome__subtitle">Mejora tu productividad centrándote en lo importante de cada momento y sé más eficiente.</h2>
      <ButtonComponent @onClick="onClick">
        Haz tu primer pomodoro
      </ButtonComponent>
    </div>
  </section>
</template>

<script>
  import ButtonComponent from '@/components/ButtonComponent'
  export default {
    name: 'Welcome',
    components:{
      ButtonComponent
    },
    methods:{
      onClick(){
        this.$router.push({name: 'pomodoro', params:{type: 'work'}})
      }
    }
  }
</script>
<style type="text/css" scoped>
  .welcome{
    background-color: var(--aplication-color-primary); 
  }

  .welcome__content
  {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    color: var(--aplication-color-text);
  }

  .welcome__title
  {
    margin-bottom: 1em;
    font-size: 2.5rem;
    font-weight: 500;
    letter-spacing: 1px;
  }

  .welcome__subtitle
  {
    font-size: 1.5rem;
    font-weight: 300;
    letter-spacing: 1px;
    margin-bottom: 1.5em;
  }
</style>
