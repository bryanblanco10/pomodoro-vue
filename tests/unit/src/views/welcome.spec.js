import { shallowMount } from '@vue/test-utils'
import Welcome from '@/views/Welcome.vue'


//Para saber que se esta renderizando bien.
describe('Welcome.vue', () => {
  it('should render property', () => {
    const welcome = shallowMount(Welcome)
    expect(welcome).toMatchSnapshot()
  })
})

describe('Methods',() => {
	it('should redirect to pomodoro route work parameter', () => {
		const routerPushSpy = jest.fn()
		const welcome = shallowMount(Welcome, {
			//mocks = this
			mocks:{
				$router:{
					push: routerPushSpy
				}
			}
		})

		welcome.vm.onClick()

		expect(routerPushSpy).toHaveBeenCalledWith({
			name: 'pomodoro',
			params:{
				type: 'work'
			}
		})
	})
})
