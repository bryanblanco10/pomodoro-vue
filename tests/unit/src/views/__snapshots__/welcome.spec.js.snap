// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome.vue should render property 1`] = `
<section class="welcome">
  <div class="aplication__content welcome__content">
    <h1 class="welcome__title">Hola, ¡bienvenido a Pomodoro!</h1>
    <h2 class="welcome__subtitle">Mejora tu productividad centrándote en lo importante de cada momento y sé más eficiente.</h2>
    <buttoncomponent-stub>
      Haz tu primer pomodoro
    </buttoncomponent-stub>
  </div>
</section>
`;
